docs/docs/kb/12-plugins.md

```diff
@@ -0,0 +1,140 @@
+---
+sidebar_position: 12
+description: Scaffolding and interacting with plugins.
+---
+
+# Plugins
+
+Ignite plugins offer a way to extend the functionality of the Ignite CLI. There are two core concepts within plugins `Plugin Commands` and `Hooks`. Where commands extend the cli's functionality, and `hooks` extend existing command functionality. Plguins can be `registered` in an Ignite scaffolded Blockchain project through the `config.yml`
+
+# Adding plugins to a project
+Plugins are registered per project. When a command is executed within a directory on disk which contains an configuration file with a plugin registered. Ignite will automatically load it into itself before a command is run. If the plugin has not already been built there will be a build step before the command is executed whcih the cli will indicate. Below is an example configuration registering a plugin:
+
+``` yaml
+plugins:
+    # must be the name same name as the plugin project name
+  - plugin:
+    # path the plugin project directory 
+    # path can be a remote git repository or local directory
+    path: /path/to/plugin
```

Comment:
I think here we should only mention the git repository, because this is what the majority of users will use. Only plugin developers will use the local directory. 

What do you think of having 2 big sections in this documentation, one for using a plugin, and an other one for developing a plugin ?

Reply:
hmm that's a good point. I'll add a section 'Developing Plugins' move this yaml example there and add a top level header here `Using Plugins`.